src/core/Akka.TestKit/TestKitBase_AwaitConditions.cs

```diff
@@ -357,34 +277,13 @@ protected static bool InternalAwaitCondition(Func<bool> conditionIsFulfilled, Ti
 
             while (!conditionIsFulfilled())
             {
-                var now = Now;
-
-                if (now > stop)
+                if(cancellationToken.IsCancellationRequested)
```

Comment:
CancellationToken check has to be done WITH the time check, do NOT delete the time elapsed check.